pkg/render/whisker/component.go

```diff
@@ -0,0 +1,427 @@
+		c.goldmaneService(),
+		c.whiskerService(),
+		c.cfg.TrustedCertBundle.ConfigMap(WhiskerNamespace))
+
+	if c.cfg.ManagementClusterConnection != nil {
+		objs = append(objs,
+			secret.CopyToNamespace(WhiskerNamespace, c.cfg.TunnelSecret)[0],
+			// TODO maybe the controller needs to create this?
+			c.cfg.TrustedCertBundle.ConfigMap(WhiskerNamespace),
+			c.guardianService(),
+		)
+	}
+	objs = append(objs, c.deployment())
+
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(WhiskerNamespace, c.cfg.PullSecrets...)...)...)
+
+	if c.cfg.ManagementClusterConnection != nil && c.cfg.TunnelSecret != nil {
+		objs = append(objs, secret.CopyToNamespace(WhiskerNamespace, c.cfg.TunnelSecret)[0])
+	}
+
+	return objs, nil
+}
+
+func (c *Component) Ready() bool {
+	return true
+}
+
+func (c *Component) serviceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta:   metav1.TypeMeta{Kind: "ServiceAccount", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: WhiskerServiceAccountName, Namespace: WhiskerNamespace},
+	}
+}
+
+func (c *Component) whiskerContainer() corev1.Container {
+	return corev1.Container{
+		Name:            WhiskerContainerName,
+		Image:           c.whiskerImage,
+		ImagePullPolicy: render.ImagePullPolicy(),
+		Env: []corev1.EnvVar{
+			{Name: "LOG_LEVEL", Value: "DEBUG"},
+			{Name: "CA_CERT_PATH", Value: c.cfg.TrustedCertBundle.MountPath()},
+		},
+		SecurityContext: securitycontext.NewNonRootContext(),
+		VolumeMounts:    c.cfg.TrustedCertBundle.VolumeMounts(rmeta.OSTypeLinux),
+	}
+}
+
+func (c *Component) whiskerService() *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "whisker",
+			Namespace: WhiskerNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{Port: 8081}},
+			Selector: map[string]string{
+				"k8s-app": WhiskerDeploymentName,
+			},
+		},
+	}
+}
+
+func (c *Component) whiskerBackendContainer() corev1.Container {
+	return corev1.Container{
+		Name:            WhiskerBackendContainerName,
+		Image:           c.whiskerBackendImage,
+		ImagePullPolicy: render.ImagePullPolicy(),
+		Env: []corev1.EnvVar{
+			{Name: "LOG_LEVEL", Value: "DEBUG"},
+			{Name: "CA_CERT_PATH", Value: c.cfg.TrustedCertBundle.MountPath()},
+			{Name: "PORT", Value: "3002"},
+			{Name: "GOLDMANE_HOST", Value: "localhost:7443"},
+		},
+		SecurityContext: securitycontext.NewNonRootContext(),
+		VolumeMounts:    c.cfg.TrustedCertBundle.VolumeMounts(rmeta.OSTypeLinux),
+	}
+}
+
+func (c *Component) goldmaneContainer() corev1.Container {
+	env := []corev1.EnvVar{
+		{Name: "LOG_LEVEL", Value: "INFO"},
+		{Name: "CA_CERT_PATH", Value: "/certs/tls.crt"},
+		{Name: "PORT", Value: "7443"},
+	}
+	var volumeMounts []corev1.VolumeMount
+	if c.cfg.LinseedPublicCASecret != nil {
+		env = append(env, corev1.EnvVar{
+			Name:  "PUSH_URL",
+			Value: fmt.Sprintf("https://%s.%s.svc/api/v1/flows/bulk", "tigera-guardian", WhiskerNamespace)})
```

Comment:
I don't think we need to string-format "tigera-guardian" sicne it's just a constant string. 

Also, can't we use localhost since these are in the same pod?

Reply:
Done.